base/ClientSweepAutomation.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+//Client sweep automation does the following - 
+//1. Starts with 50 threads and increments by 50 for every iteration. Each iteration is run for 10 seconds.
+//2. A % difference is calculated between the TPS of the current and previous iteration.
+//3. When there is a % difference of <= 1% for 7 consecutive iterations, the sweep is stopped (indicating a plateau). You can change this number 7 using the const variable plateau.
+//4. As the sweep progresses, the client threads corresponding to peak TPS values are saved. When there is more than 2% difference with a previous peak, a new peak is determined.
+//5. The sweep can go up to 20 iterations if the optimal number of client threads is not deduced earlier. You can change this number 20 using the const client_sweep_max_iterations.
+
+spl_autoload_register(function ($class_name){
+	include $class_name . '.php';
+});
+
+class ClientSweepAutomation {
+    const client_sweep_max_iterations = 20;
+    const plateau = 7;
+    private $curr_tps;
+    private $prev_tps;
+    private $max;
+    private $failedReq;
+    private $diff_percent;
+    private $plateau_tracker;
+    private $clientsweep_results;
+    private $optimalClientThreads;
+
+    public function __construct(){
+        $this->curr_tps = 1;
+        $this->prev_tps = 0;
+        $this->max = 0;
+        $this->failedReq = 0;
+        $this->diff_percent = 0;
+        $this->plateau_tracker = 0;
+        $this->clientsweep_results = array();
+        $this->optimalClientThreads = 0;
+    }
+
+    public function GetOptimalThreads(PerfOptions $options, PerfTarget $target){
+        $options->clientThreads = 0;
+        for($iter=1; $iter<=self::client_sweep_max_iterations; $iter++){
```

Comment:
Please change iter to something more meaningful since it's used throughout the file.

Reply:
Done